test/helpers/ssh_command.go

```diff
@@ -208,10 +208,23 @@ func (client *SSHClient) RunCommandContext(ctx context.Context, cmd *SSHCommand)
 	}
 	session.RequestPty("xterm-256color", 80, 80, modes)
 
+	stdin, err := session.StdinPipe()
+	if err != nil {
+		log.Errorf("Could not get stdin", err)
+	}
+
 	go func() {
 		select {
 		case <-ctx.Done():
-			if err := session.Signal(ssh.SIGHUP); err != nil {
+			_, err := stdin.Write([]byte{3})
+			if err != nil {
+				log.Errorf("write ^C error:", err)
+			}
+			err = session.Wait()
+			if err != nil {
+				log.Errorf("wait error:", err)
+			}
+			if err = session.Signal(ssh.SIGHUP); err != nil {
```

Comment:
I've been thinking about this, and worse comes to worse, if sending signals over the SSH session doesn't work, we could try to kill it by doing a `kubectl exec <kill microscope>` in the cilium pod? :/